### DISCUSSION TITLE
Line by line  debugging async scenarios

### DISCUSSION THREAD

**Discussion by mrcjbf0001** [codeceptjs/CodeceptJS#2963] [Q&A]
I am looking for the best way, or any possible way really, to step through an async scenario line by line.  Right now I can set break points, open a gitbash javascript debugger terminal from within vs code, and launch my test scenario with the debugger attached.  When I step over a line or continue the debugger to the next break point, the actions are not executed in the browser, except for console.log statements.  once every line in the scenario has been executed, the actions will all happen in sequence in the browser.  I am used to debugging java code where each line is executed when you click the step over button from the debugger.  

this question was asked in 2018, and the solution given was to add the await command before each line.  This does work, but does anyone know of a way to do it without adding the await command?

https://github.com/codeceptjs/CodeceptJS/issues/1144

**Comment by VoidEssy** [comment]
You need to add await to every line it's in the documentation here I believe: http://xt1.org/codecept/debugging/ at the bottom of this page :)
